tasks/xgettext.js

```diff
@@ -319,48 +370,55 @@ module.exports = function(grunt) {
+        var newMessageToFilesMap = newExtracted[1];
         _.each(newExtractedStrings, function(aNamespace, namespaceName) {
+          if (!messageToFilesMap[namespaceName]) {
+            messageToFilesMap[namespaceName] = {};
+          }
+          combineMessageToFilesMaps(messageToFilesMap[namespaceName], newMessageToFilesMap[namespaceName]);
           messages[namespaceName] = _.extend(messages[namespaceName] || {}, aNamespace);
         });
       });
 
       mergeTranslationNamespaces(translations, messages);
-
-      _.each(messages, function(aNamespace, namespaceName) {
-        var count = Object.keys(aNamespace).length;
-      })
     });
-
+    var errors = [];
     _.each(translations, function(aNamespace, namespaceName) {
-      var contents = "";
-
-      var sortedKeys = Object.keys(aNamespace).sort();
-
-      contents += _.map(sortedKeys, function (aKey) {
-        var definition = aNamespace[aKey];
-        var buffer = "msgid " + escapeString(definition.singular) + "\n";
-        if (definition.plural) {
-          buffer += "msgid_plural " + escapeString(definition.plural) + "\n";
-          buffer += "msgstr[0] " + escapeString(definition.message) + "\n";
-        } else {
-          buffer += "msgstr " + escapeString(definition.message) + "\n";
-        }
-        return buffer;
-      }).join("\n");
 
       var filename = path.resolve(options.potPath, namespaceName + ".pot");
 
       if (!grunt.option('fix')) {
-        var existingResources = grunt.file.read(filename);
-        if (contents !== existingResources) {
-          grunt.fail.fatal("It appears that you have js resource changes not yet updated in po and pot files. Please run 'grunt xgettext --fix'");
+        var existingString = grunt.file.read(filename);
+        var existingMessageIds = extractMessageIds(existingString);
+        var diff = calculateDiffBetweenOldAndNewMessageIds(existingMessageIds, messageToFilesMap[namespaceName]);
+        if (diff) {
+          errors.push(diff);
         }
       } else {
+        var contents = "";
+
+        var sortedKeys = Object.keys(aNamespace).sort();
+
+        contents += _.map(sortedKeys, function (aKey) {
+          var definition = aNamespace[aKey];
+          var buffer = "msgid " + escapeString(definition.singular) + "\n";
+          if (definition.plural) {
+            buffer += "msgid_plural " + escapeString(definition.plural) + "\n";
+            buffer += "msgstr[0] " + escapeString(definition.message) + "\n";
+          } else {
+            buffer += "msgstr " + escapeString(definition.message) + "\n";
+          }
+          return buffer;
+        }).join("\n");
         grunt.file.write(filename, contents, {});
         updatePoFromPot(potFolderPath, namespaceName);
       }
     });
+    if (errors.length) {
+      grunt.fail.fatal("It appears that you have js resource changes not yet updated in po and pot files. Please run 'grunt xgettext --fix'. The diff is:\n" + errors.join('\n'));
```

Comment:
"js resource changes" is a bit confusing, it seems to indicate some other sort of file rather than textual resources defined in a JavaScript file.

Reply:
yeah the message is kind of bad, I should change it